.jenkins/pipelines/standalone/azure-sdk-tests.Jenkinsfile

```diff
@@ -27,20 +28,36 @@ pipeline {
     stages {
         stage("Cleanup files") {
             steps {
-                sh """
-                   ${JENKINS_SCRIPTS}/global/clean-temp.sh
-                   """
+                sh "${JENKINS_SCRIPTS}/global/clean-temp.sh"
             }
         }
-        stage('Verify commit sync') {
-            when { allOf {
-                expression { params.COMMIT_SYNC != "" }
-                expression { params.COMMIT_SYNC != GIT_COMMIT }
-            }}
+        stage("Checkout Pull Request") {
+            when {
+                expression { params.PULL_REQUEST_ID != "" }
+            }
             steps {
+                cleanWs()
+                checkout([$class: 'GitSCM',
+                    branches: [[name: "pr/${PULL_REQUEST_ID}"]],
+                    extensions: [],
+                    userRemoteConfigs: [[
+                        url: 'https://github.com/deislabs/mystikos',
+                        refspec: "+refs/pull/${PULL_REQUEST_ID}/merge:refs/remotes/origin/pr/${PULL_REQUEST_ID}"
+                    ]]
+                ])
+                // Check if the checked out commit is the same across all parallel builds
                 script {
-                    currentBuild.result = 'ABORTED'
-                    error("Aborting build: mismatched commit - commit($GIT_COMMIT), expected(${COMMIT_SYNC})")
+                    if ( COMMIT_SYNC != "" ) {
```

Comment:
I think this check should be in a different stage, otherwise the commit sync check won't happen for nightly/code coverage builds. Likewise for other tests

Reply:
Good catch!